### ISSUE TITLE
Support image classification tasks

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/AutoMONAI#29]
## Motivation

AutoMONAI currently only supports segmentation (pixel-level labels → Dice/IoU metrics). Many medical imaging tasks are classification: disease detection, grading, staging, etc. The core architecture (model factory, training loop, transforms, UI) can be extended to support classification with relatively contained changes.

## Current assumptions that are segmentation-specific

| Component | Segmentation assumption |
|---|---|
| `models.py` | All models output spatial maps (B, C, H, W) |
| `train.py` | Loss expects spatial predictions; metrics are Dice/IoU |
| `dataset.py` | Labels are image files (masks) loaded from `labelsTr/` |
| `inference.py` | Output is a spatial prediction saved as an image |
| `results.py` | Metrics CSV assumes segmentation metric names |
| UI Generate tab | Only segmentation models, losses, and metrics listed |

### Task mode
```
--task segmentation | classification (default: segmentation)
```

This flag controls which models, losses, metrics, and dataset loading logic are used.

### Models (`models.py`)
Add classification heads / models:
- **Existing MONAI models with classification head**: DenseNet121, DenseNet169, DenseNet201, EfficientNetBN, SEResNet50, SEResNext50, ResNet (10/18/34/50/101/152/200), ViT
- Model factory returns `(B, num_classes)` output for classification, `(B, C, H, W)` for segmentation
- `get_model(name, task="classification", num_classes=2, ...)`

### Dataset (`dataset.py`)
- **Classification labels**: single integer or one-hot per image instead of a mask file
- Support label source:
  - `labels.csv` — CSV with `filename,label` columns in the dataset directory
  - `dataset.json` — nnUNet-style metadata extended with class labels
  - Folder structure — `imagesTr/class_0/`, `imagesTr/class_1/`
- `get_train_files()` returns `{"image": path, "label": int}` for classification

### Losses
Add classification losses:
- Cross-entropy (already exists but currently used for segmentation)
- Binary cross-entropy
- Focal loss (classification variant)
- Label smoothing cross-entropy
- AUC loss

### Metrics
Add classification metrics:
- Accuracy, balanced accuracy
- AUC (ROC), AUPRC
- F1 / precision / recall (per-class and macro)
- Confusion matrix
- Cohen's kappa

### Training loop (`train.py` / `run.py`)
- Skip spatial post-processing (argmax over spatial dims) for classification
- Use `softmax` → class prediction for classification
- Validation: report accuracy/AUC instead of Dice

### Inference (`inference.py`)
- Classification inference outputs predicted class + confidence scores (not spatial maps)
- Results saved as CSV: `filename, predicted_class, confidence, ...per_class_probs`

### UI changes
- **Task selector** at the top of Generate tab: `Segmentation` / `Classification`
- Swap model dropdown, loss dropdown, and metrics checkboxes based on task
- Classification-specific: class count input, class names input
- Results display: confusion matrix, ROC curve (if we add charting, or via W&B)

### CLI
```
--task              segmentation | classification (default: segmentation)
--num_classes       Number of classes for classification (default: 2)
--label_source      csv | json | folder (default: csv)
--label_file        Path to labels CSV/JSON (default: labels.csv in dataset dir)
```

## Out of scope
- Multi-label classification (multiple labels per image) — follow-up issue
- Object detection / bounding boxes — different paradigm entirely
- Regression tasks (e.g. age prediction) — follow-up issue
